src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -268,85 +371,73 @@ protected void setUpDatabase() throws SQLException {
                 + "payloadFormat INTEGER,"
                 + "contentType STRING"
                 + ");";
-        DriverManager.registerDriver(new org.sqlite.JDBC());
-        performSqlOperation(conn -> {
-            try (Statement st = conn.createStatement()) {
-                // create new table if table doesn't exist
-                st.executeUpdate(query);
-                return null;
-            }
-        });
+
+        @Override
+        protected PreparedStatement createStatement(Connection conn) throws SQLException {
+            return conn.prepareStatement(QUERY);
+        }
+
+        @Override
+        protected Integer doExecute(PreparedStatement statement) throws SQLException {
+            return statement.executeUpdate();
+        }
+
+        @Override
+        public Void onNewConnection(Connection newConnection) throws SQLException {
+            super.onNewConnection(newConnection);
+            execute(); // recreate spooler table right away
+            return null;
+        }
     }
 
-    @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    private <T> T performSqlOperation(CrashableFunction<Connection, T, SQLException> operation) throws SQLException {
-        try {
+    abstract class CachedStatement<T extends Statement, R> {
+        private T statement;
+        private Connection connection;
```

Comment:
connect looks unused

Reply:
Ah yeah, field is leftover from before using callbacks, will remove next time I'm at my laptop

Reply:
fixed